pallets/xyk/src/lib.rs

```diff
@@ -595,6 +607,36 @@ pub mod pallet {
 
 			<Self as XykFunctionsTrait<T::AccountId>>::promote_pool(liquidity_token_id)
 		}
+
+		#[pallet::weight(10_000)]
+		pub fn activate_liquidity(
+			origin: OriginFor<T>,
+			liquidity_token_id: TokenId,
+			amount: Balance,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			<Self as XykFunctionsTrait<T::AccountId>>::activate_liquidity(
+				sender,
+				liquidity_token_id,
+				amount,
+			)
+		}
```

Comment:
I suggest you use the transactional annotation, like Mat has used in the bootstrap pallet.
You should use it for all extrinsics, EXCEPT for buy and sell asset as they need to charge percent for every txn even if it fails.